### ISSUE TITLE
Feature Request: Subdivision of the table components. Like spliting a column into two.

### ISSUE THREAD

**Issue by PranavBhattarai** [giuspen/cherrytree#2340]
Currently can create:
 - Rows
 - Column
 

| Name         | Age               |            Sex             |
| ----------------|------------------ | ------------------------|
| Bond          | 38                  | Male                      |
 
 But we cannot further split the column if we want to.

 Like a **Name** column split into two, **First Name** and **Surname**.
 It would be really handy if we could split a column by two at least.

**Comment by PranavBhattarai** [comment]
When we start to play in Tables, it becomes almost impossible to not see the need for this. 
It is more about **categorizing** the table in a more orderly fashion. 

> add a column instead of splitting an existing one.

Yeah, that is the basic least approach that is available right now. 

Let me give you a more complex example:
![image](https://github.com/giuspen/cherrytree/assets/32763112/614b259e-bf29-4e6c-9bb7-8321408fe229)
 
 It makes more sense to give more control while designing the table, isn't it? 
 Currently, we can't unbold, table headings, colorize the texts inside the table,  italicize any words inside the table, etc.
 The table we have is limited control. 
 
 It would be great if we could open up some control that goes into designing the table.

**Comment by Sisyfos3** [comment]
@PranavBhattarai , I agree on formatting the cell contents (bold/unbold, cell color, italic etc), but splitting a column or cell can't be done even in Excel, which seems to be the world standard (I'm not particularly fond of it), unless cells/column are joined to begin with. I'm not a programmer so I have no clue as to how much coding would be neccesary to make splitting cells/columns possible.
